core/src/data.rs

```diff
@@ -223,6 +244,16 @@ impl Into<i64> for DataType {
     }
 }
 
+impl Into<i32> for DataType {
+    fn into(self) -> i32 {
+        if let DataType::Int(s) = self {
+            s
+        } else {
+            unreachable!();
+        }
+    }
+}
```

Comment:
Separately from this PR, I wish we had a nicer way to deal with all these `DataType` conversions. We may want a macro that generates `TryFrom` and `TryInto` conversions.